src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -17,7 +16,6 @@ public class SocketInput : ICriticalNotifyCompletion
 
         private readonly MemoryPool2 _memory;
         private readonly IThreadPool _threadPool;
-        private readonly ManualResetEventSlim _manualResetEvent = new ManualResetEventSlim(false);
```

Comment:
Removing this seems scary. I think this breaks reasonable assumptions about how GetResult() works.